File: January/30- N-th Tribonacci Number/30- N-th Tribonacci Number (Noura Algohary).cpp
// Author: Noura Algohary
class Solution {
public:
    int tribonacci(int n) {
        int t0 = 0, t1 = 1, t2 = 1, t3 = 0;

        if(n == 0)
            return 0;
        else if(n == 1 || n == 2)
            return 1;
        
        for(int i=3;i<=n;i++)
        {
            t3 = t0 + t1 + t2;
            t0 = t1;
            t1 = t2;
            t2 = t3;
        }
        
        return t3;
    }
};

File: 01- January/30- N-th Tribonacci Number/30- N-th Tribonacci Number (Noura Algohary).py
# Author: Noura Algohary
class Solution:
    def tribonacci(self, n: int) -> int:
        t0, t1, t2 = 0, 1, 1

        if n < 2:
            return n
        elif n == 2:
            return 1
        
        for i in range(2, n):
            t3 = t0 + t1 + t2
            t0 = t1
            t1 = t2
            t2 = t3
        
        return t3